cmd/csender/main.go

```diff
@@ -0,0 +1,157 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/cloudradar-monitoring/cagent/pkg/csender"
+)
+
+var (
+	// set on build:
+	// go build -o csender -ldflags="-X main.version=$(git --git-dir=src/github.com/cloudradar-monitoring/cagent/.git describe --always --long --dirty --tag)" github.com/cloudradar-monitoring/cangent/cmd/csender
+	version string
+)
+
+type boolFlag struct {
+	set   bool
+	value bool
+}
+
+func (sf *boolFlag) Set(value string) error {
+	i, err := strconv.Atoi(value)
+	if err != nil {
+		return err
+	}
+
+	switch i {
+	case 0:
+		sf.value = false
+	case 1:
+		sf.value = true
+	default:
+		return fmt.Errorf("arg can be either 0 or 1")
+	}
+
+	sf.set = true
+	return nil
+}
+
+func (sf *boolFlag) String() string {
+	var s string
+	if sf.set {
+		if sf.value {
+			s = "1"
+		} else {
+			s = "0"
+		}
+	}
+
+	return s
+}
+
+func (sf *boolFlag) BoolPtr() *bool {
+	if !sf.set {
+		return nil
+	}
+
+	return &sf.value
+}
+
+func fatal(msg string) {
+	_, _ = fmt.Fprintln(os.Stderr, msg)
+	os.Exit(1)
+}
+
+func main() {
+	var successFlag boolFlag
+
+	checkNamePtr := flag.String("n", "", "check name (*required)")
+	tokenPtr := flag.String("t", "", "custom check token (*required)")
+	hubURLPtr := flag.String("u", "https://hub.cloudradar.io/cct/", "hub URL to use")
+	flag.Var(&successFlag, "s", "set success [0,1]")
+	alertMessagePtr := flag.String("a", "", "alert message")
+	warningMessagePtr := flag.String("w", "", "warning message")
+
+	versionPtr := flag.Bool("version", false, "show the csender version")
+
+	flag.Parse()
+
+	if *versionPtr {
+		fmt.Printf("csender v%s released under MIT license. https://github.com/cloudradar-monitoring/csender/\n", version)
```

Comment:
The repo address now is `cloudradar-monitoring/cagent`

Reply:
fixed